scripts/run_integration_tests_vfio.sh

```diff
@@ -7,9 +7,32 @@ source $(dirname "$0")/test-util.sh
 process_common_args "$@"
 
 WORKLOADS_DIR="$HOME/workloads"
-FOCAL_OS_IMAGE_NAME="focal-server-cloudimg-amd64-custom-20210609-0.raw"
-FOCAL_OS_IMAGE="$WORKLOADS_DIR/$FOCAL_OS_IMAGE_NAME"
+mkdir -p "$WORKLOADS_DIR"
+
+FW_URL=$(curl --silent https://api.github.com/repos/cloud-hypervisor/rust-hypervisor-firmware/releases/latest | grep "browser_download_url" | grep -o 'https://.*[^ "]')
 FW="$WORKLOADS_DIR/hypervisor-fw"
+if [ ! -f "$FW" ]; then
+    pushd $WORKLOADS_DIR
+    time wget --quiet $FW_URL || exit 1
+    popd
+fi
+
+FOCAL_OS_IMAGE_NAME="focal-server-cloudimg-amd64-custom-20210609-0.qcow2"
+FOCAL_OS_IMAGE_URL="https://cloud-hypervisor.azureedge.net/$FOCAL_OS_IMAGE_NAME"
+FOCAL_OS_IMAGE="$WORKLOADS_DIR/$FOCAL_OS_IMAGE_NAME"
+if [ ! -f "$FOCAL_OS_IMAGE" ]; then
+    pushd $WORKLOADS_DIR
+    time wget --quiet $FOCAL_OS_IMAGE_URL || exit 1
+    popd
+fi
+
+FOCAL_OS_RAW_IMAGE_NAME="focal-server-cloudimg-amd64-custom-20210609-0.raw"
+FOCAL_OS_RAW_IMAGE="$WORKLOADS_DIR/$FOCAL_OS_RAW_IMAGE_NAME"
+if [ ! -f "$FOCAL_OS_RAW_IMAGE" ]; then
+    pushd $WORKLOADS_DIR
+    time qemu-img convert -p -f qcow2 -O raw $FOCAL_OS_IMAGE_NAME $FOCAL_OS_RAW_IMAGE_NAME || exit 1
+    popd
+fi
```

Comment:
You should also check the integrality of  images like what is done here [1][2].

[1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/scripts/run_integration_tests_x86_64.sh#L21
[2] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/scripts/run_integration_tests_x86_64.sh#L99-L104.

Reply:
Not all images are downloaded for the vfio test, so same sha1sum check would cause errors and sha1sum will return error code 1

```
$ sha1sum -c sha1sums-x86_64 
sha1sum: alpine-minirootfs-x86_64.tar.gz: No such file or directory
alpine-minirootfs-x86_64.tar.gz: FAILED open or read
focal-server-cloudimg-amd64-custom-20210609-0.qcow2: OK
focal-server-cloudimg-amd64-custom-20210609-0.raw: OK
sha1sum: jammy-server-cloudimg-amd64-custom-20221118-1.qcow2: No such file or directory
jammy-server-cloudimg-amd64-custom-20221118-1.qcow2: FAILED open or read
sha1sum: jammy-server-cloudimg-amd64-custom-20221118-1.raw: No such file or directory
jammy-server-cloudimg-amd64-custom-20221118-1.raw: FAILED open or read
sha1sum: WARNING: 3 listed files could not be read
```

Solve this by using --ignore-missing argument
```
$ sha1sum --ignore-missing -c sha1sums-x86_64 
focal-server-cloudimg-amd64-custom-20210609-0.qcow2: OK
focal-server-cloudimg-amd64-custom-20210609-0.raw: OK
```